main.c

```diff
@@ -753,8 +753,16 @@ static void *tcmur_cmdproc_thread(void *arg)
 	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
 	struct tcmur_device *rdev = tcmu_dev_get_private(dev);
 	struct pollfd pfd;
-	int ret;
 	bool dev_stopping = false;
+	char *pname;
+	int ret;
+
+	if (asprintf(&pname, "cmdproc-%s", tcmu_dev_get_td_name(dev)) == -1) {
```

Comment:
The man page for pthread_setname_np states that the name can't be more than 16 chars including the terminating null. The tcmu lio device names are probably limited by what configfs can support and that seems to be the normal linux name length.

So maybe we want to use a different identifier.

Reply:
The hba name length range will be [6, 25], so this still won't work.

```
# mkdir /sys/kernel/config/target/core/user_12345678901234567890
# mkdir /sys/kernel/config/target/core/user_123456789012345678901
mkdir: cannot create directory ‘/sys/kernel/config/target/core/user_123456789012345678901’: Numerical result out of range

```

The dev_name will be the okay, it will be "uio%d" and the "%d" is from idev->minor.

```
 69 struct uio_device {            
 70         struct module           *owner;     
 71         struct device           *dev;       
 72         int                     minor;                                                                                                                                 
 73         atomic_t                event;      
 74         struct fasync_struct    *async_queue;
 75         wait_queue_head_t       wait;       
 76         struct uio_info         *info;      
 77         struct kobject          *map_dir;   
 78         struct kobject          *portio_dir;
 79 };

```
But the minor max length will be 1048576, which if from:

#define UIO_MAX_DEVICES         (1U << MINORBITS)
#define MINORBITS 20

In Arch 64, the "uio%d" length will range in [4, 11]